### PR TITLE
Refactor file preferences

### PR DESCRIPTION
**Pull request by calixtus**
This PR is going to refactor some parts of the code in relationship with the FilePreferences. It's going to be rather big again, since FilePreferences and other Preferences objects are not clearly distinguishable and this will take probably some time for me to contemplate how to resort them.

Work in progress, but here for you to preview and to comment, if you have some good ideas.

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by calixtus** [comment]
grafik
grafik

File: src/main/java/org/jabref/gui/preferences/LinkedFilesTabViewModel.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        searchFilesOnOpenProperty.setValue(initialFilePreferences.shouldSearchFilesOnOpen());
+        openBrowseOnCreateProperty.setValue(initialFilePreferences.shouldOpenBrowseOnCreate());
+        fileNamePatternProperty.setValue(initialFilePreferences.getFileNamePattern());
+        fileDirectoryPatternProperty.setValue(initialFilePreferences.getFileDirectoryPattern());
+
+        // Autolink preferences
+        switch (initialAutoLinkPreferences.getCitationKeyDependency()) {
+            case START -> autolinkFileStartsBibtexProperty.setValue(true);
+            case EXACT -> autolinkFileExactBibtexProperty.setValue(true);
+            case REGEX -> autolinkUseRegexProperty.setValue(true);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/preferences/LinkedFilesTabViewModel.java:86]: default missing?

**Comment by mlep** [comment]
@calixtus 
I am a bit puzzled by the new `import and export` window: for me, its content is not about importing and exporting a library, but about the actions taking place when opening and saving a library.
If this content is called import and export, I guess it will be confused with custom import and export.

Note: the `file reformatting` section use both `save` and `export`, adding to my confusion. My guess is that only `save` should be used.

**Comment by calixtus** [comment]
grafik

**Comment by calixtus** [comment]
@mlep I reworded the tab again to FileTab.
We talked here at JabCon about the difference between SaveSortOrder in LibraryProperties and ExportSortOrder in the PreferencesTab: ExportSortOrder in the Preferences takes effect probably in the custom exporters, while SaveSortOrder has only effect on saving the bib file.

**Comment by mlep** [comment]
@calixtus Thank for clarifying.
So, if we go with " `Options -> Preferences` is for default behavior, and `Library -> Library properties` menu is for a specific file " (as I wrote here: https://docs.jabref.org/setup), I am wondering where to define the default saving order for the library in the `Preferences`.

In the future, will `Options -> Manage custom exports` and `Options -> Manage custom imports` integrate the `preferences` window? In such a case, an `Import and export` page in the `Preferences` will need to be created (and will likely contain the `Export sort order`).

Happy JabCon20!

**Comment by calixtus** [comment]
Yes, thanks for that thought. Refactoring the preferences is a lengthy process I'm working on since some time now and especially here I was pretty clueless about how to reorder them.
It's exactly as you have described it, the options-menu should present all the general and default options for jabref, the library menu the library-specific options which override the general options.
Obviously, the default SaveSortOrder was abolished in favor of an alphabetical SortOrder, which can be adjusted for every single library via library properties, some time ago.
My plan is to integrate all the options dialogues into one grup-d and also in the more further future to integrate all the library properties dialogs into an unified library properties dialog, so we can abolish the "options" menu at all and move the options->preferences menucommand into another menu (like file).

**Comment by mlep** [comment]
Thank you for your explanation.
I am recognizing your huge effort in improving JabRef and do not want to be disturbing.

**Comment by calixtus** [comment]
> Thank you for your explanation.
> I am recognizing your huge effort in improving JabRef and do not want to be disturbing.

The opposite is the case, I'm thankful for critics, feedback and people thinking about the future in this matter.

**Pull request merged**